api_types.ts

```diff
@@ -559,8 +559,8 @@ export interface CommonPackFormulaDef<T extends ParamDefs> {
    *
    * @example
    * ```
-   * validateParameters: async function (context, _, formulaContext) {
-   *   let [quantity, sku] = formulaContext?.params;
+   * validateParameters: async function (context, _, params) {
+   *   let {quantity, sku} = ensureExists(params);
```

Comment:
this ensureExists is because metadataContext may be undefined for metadata formulas even though it should never be for validateParameters

Reply:
Not for this PR but can you check whether we're always passing a metadata context into all metadata formulas in practice? Likely just `{}`? If so we could just update the type to not be optional and this would be much more ergonomic. Or I guess we could just make it so. Otherwise we should update the ContextT here somehow so that in this case it's not optional. This isn't very ergonomic otherwise.

Reply:
Noting it down to look into this